packages/network-controller/src/NetworkController.ts

```diff
@@ -429,6 +455,77 @@ export class NetworkController extends BaseControllerV2<
     }
     return Promise.resolve(true);
   }
+
+  /**
+   * Network Configuration management functions
+   */
+
+  /**
+   * Adds a network configuration if the rpcUrl is not already present on an
+   * existing network configuration. Otherwise updates the entry with the matching rpcUrl.
+   *
+   * @param config - The network configuration.
+   * @param config.rpcUrl - The network configuration RPC URL.
+   * @param config.chainId - The chain ID of the network, as per EIP-155.
+   * @param config.ticker - Currency ticker.
+   * @param config.nickname - Personalized network name.
+   * @param config.rpcPrefs - Personalized preferences.
+   * @returns uuid for the added or updated network configuration
+   */
+  upsertNetworkConfiguration({
+    rpcUrl,
+    chainId,
+    ticker,
+    nickname,
+    rpcPrefs,
+  }: NetworkConfiguration): string {
+    const networkConfigurations = { ...this.state.networkConfigurations };
+
+    const newNetworkConfiguration: NetworkConfiguration = {
+      rpcUrl,
+      chainId,
+      ticker,
+      nickname,
+      rpcPrefs,
+    };
+
+    for (const configUUID in networkConfigurations) {
+      if (
+        networkConfigurations[configUUID].rpcUrl.toLowerCase() ===
+        rpcUrl.toLowerCase()
+      ) {
+        // update existing config
+        networkConfigurations[configUUID] = newNetworkConfiguration;
+        this.update((state) => {
+          state.networkConfigurations = networkConfigurations;
+        });
```

Comment:
Because `networkConfiguration` is pulled out of state, this line will update state directly, making the call to `this.update` redundant. What do you think about:

``` suggestion
        this.update((state) => {
          state.networkConfigurations[configUUID] = newNetworkConfiguration;
        });
```

Reply:
done [here](https://github.com/MetaMask/core/pull/1064/commits/ac392e90922842cc69a7cb6a7377557c532dbc35)